contracts/Orchestrator.sol

```diff
@@ -22,6 +22,8 @@ contract Orchestrator is Ownable {
     // Stable ordering is not guaranteed.
     Transaction[] public transactions;
 
+    address[] private whitelist;
```

Comment:
Maybe use `EnumerableSet` here, so we don't have to loop through the list to check?